src/types/SAVTypes/cfru/PK3CFRU.ts

```diff
@@ -0,0 +1,527 @@
+
+      // EVs 43:49
+      this.evs = readStatsFromBytes(dataView, 0x2c)
+
+      // 49
+      this.pokerusByte = dataView.getUint8(0x32)
+
+      // 50
+      this.metLocationIndex = dataView.getUint8(0x33)
+      this.internalMetLocationIndex = this.metLocationIndex
+
+      // 51:53
+      this.metLevel = uIntFromBufferBits(dataView, 0x34, 0, 7, true)
+
+      this.gameOfOrigin = uIntFromBufferBits(dataView, 0x34, 7, 4, true)
+      this.internalGameOfOrigin = this.gameOfOrigin
+
+      // OHPKM will not copy this field if the mon was already being tracked before being
+      // transferred to Radical Red
+      this.pluginOrigin = this.gameOfOrigin === INTERNAL_ORIGIN_FROM_RR ? 'radical_red' : undefined
+
+      this.canGigantamax = getFlag(dataView, 0x34, 11)
+      this.trainerGender = getFlag(dataView, 0x34, 15)
+
+      // 53:57
+      this.ivs = read30BitIVsFromBytes(dataView, 0x36)
+      this.isEgg = getFlag(dataView, 0x36, 30)
+      this.hasHiddenAbility = getFlag(dataView, 0x36, 31)
+    } else {
+      const other = arg
+
+      this.personalityValue = generatePersonalityValuePreservingAttributes(other) ?? 0
+      this.trainerID = other.trainerID
+      this.secretID = other.secretID
+      this.languageIndex = other.languageIndex
+      this.markings = markingsFourShapesFromOther(other.markings) ?? {
+        circle: false,
+        triangle: false,
+        square: false,
+        heart: false,
+      }
+      this.dexNum = other.dexNum
+      this.formeNum = other.formeNum
+      this.privateHeldItemIndex = this.itemFromString(other.heldItemName)
+      this.heldItemIndex = ItemFromString(other.heldItemName)
+      this.exp = other.exp
+      this.movePPUps = other.movePPUps
+      this.trainerFriendship = other.trainerFriendship ?? 0
+      this.moves = other.moves.map(this.moveToGameIndex).map(this.moveFromGameIndex)
+
+      for (let i = 0; i < 4; i++) {
+        const pp = getMoveMaxPP(this.moves[i], this.format, this.movePPUps[i])
+
+        if (pp) this.movePP[i] = pp
+      }
+      this.evs = other.evs ?? {
+        hp: 0,
+        atk: 0,
+        def: 0,
+        spe: 0,
+        spa: 0,
+        spd: 0,
+      }
+      this.pokerusByte = other.pokerusByte ?? 0
+      this.metLocationIndex = other.metLocationIndex ?? 0
+      this.metLevel = other.metLevel ?? 0
+
+      const fromRadicalRed = other.pluginOrigin === 'radical_red'
+
+      if (fromRadicalRed) {
+        this.gameOfOrigin = INTERNAL_ORIGIN_FROM_RR
+        this.internalGameOfOrigin = this.gameOfOrigin
+
+        this.metLocationIndex = other.metLocationIndex ?? FIRERED_IN_GAME_TRADE
+        this.internalMetLocationIndex = this.metLocationIndex
+      } else {
+        this.gameOfOrigin = other.gameOfOrigin ?? INTERNAL_ORIGIN_NON_RR
+        this.internalGameOfOrigin = INTERNAL_ORIGIN_NON_RR
+
+        this.metLocationIndex = other.metLocationIndex ?? FIRERED_IN_GAME_TRADE
+        this.internalMetLocationIndex = FIRERED_IN_GAME_TRADE
+      }
+
```

Comment:
I would move this code out of the super class and into the subclasses, after a `super()` call. This is why you're having issues tracking the `unbound` origin